enumerable.rb

```diff
@@ -1,72 +1,130 @@
-        result
-    end
-
-    def my_all
-        result = true
-        my_each do |x|
-            result = false unless yield x
-        end
-        result
-    end
-
-    def my_any?
-        result = false
-        my_each do |x|
-            result = false if yield x
-        end 
-        result
-    end
-
-    def my_none?
-        result = true
-        my_each do |x|
-            result = false if yield x
-        end
-        result
-    end
-
-    def my_count
-        count = 0
-        my_each do |x|
-            count +=1 if yield x
-        end
-        count
-    end
-
-    def my_map(arr)
-        result = []
-        my_each(arr) do |x|
-            result.push(yield(x))
-        end
-        result
-    end
-
-
-
-
-
-
-
-end
\ No newline at end of file
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+    end
+
+  def my_each_index
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    end
+
+  def my_select
+    result = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |x|
+      result.push(x) if yield x
+    end
+    result
+  end
+
+  def my_all?
+    if block_given?
+      my_each do |x|
+        false unless yield x
+      elsif val.nil?
+        my_each do |x|
+          false unless x
+        elsif val.class == Class
+          my_each do |x|
+            false unless x.is_a? val
+          elsif val.class == Regexp
+            my_each do |x|
+              false unless val.match? element
+            else
+              my_each do |x|
+                false unless element == val
+              end
+              true
+            end
+
+
+  def my_any?
+    if block_given?
+      my_each do |x|
+        false unless yield x
+      elsif val.nil?
+        my_each do |x|
+          false unless x
+        elsif val.class == Class
+          my_each do |x|
+            false unless x.is_a? val
+          elsif val.class == Regexp
+            my_each do |x|
+              false unless val.match? element
+            else
+              my_each do |x|
+                false unless element == val
+              end
+              true
+            end
+
+  def my_none?
+    if block_given?
+      my_each do |x|
+        false unless yield x
+      elsif val.nil?
+        my_each do |x|
+          false unless x
+        elsif val.class == Class
+          my_each do |x|
+            false unless x.is_a? val
+          elsif val.class == Regexp
+            my_each do |x|
+              false unless val.match? element
+            else
+              my_each do |x|
+                false unless element == val
+              end
+              true
+            end
+
+  def my_count(_val = nil)
+    if count.zero?
+      my_each do |x|
+      count += 1 if yield x
+      end
+      count
+    end
```

Comment:
:red_circle: **ATTENTION** 

You are not closing your statements correctly!! This is a huge problem cause nothing will run properly!!
You need to fix this issue first. Please check you have all the `end` where they should be.